### PR TITLE
Add support for resource mapping in the Stackdriver exporter config

### PR DESCRIPTION
**Pull request by nilebox**
**Description:** 
Adding a feature - declarative config for mapping OpenCensus resources to Stackdriver resources.

**Testing:** Unit tests

**Documentation:** -

### CODE REVIEW

**Review by rghetia — COMMENTED** [review]
minimum label check functionality is missing. For example, for a specific resource_type stackdriver expects that certain labels be present. Even if one of the label is not present then entire time-series request is rejected.
So, either all required mappings are mandatory OR create additional parameter for each label whether it is mandatory or not. If it is mandatory and the label is missing then it defaults to defaultMapper.

**Comment by nilebox** [comment]
> So, either all required mappings are mandatory OR create additional parameter for each label whether it is mandatory or not.

Made labels required by default, with extra `optional` parameter to make some of them optional.

File: exporter/stackdriverexporter/resource_mapper_test.go [hunk 1]
```diff
@@ -38,45 +43,78 @@ func TestResourceMapper(t *testing.T) {
...
 			},
+			// Resource with missing required labels should be converted via default implementation
 			wantResource: &monitoredres.MonitoredResource{
 				// Resource type transformed
-				Type: "target_resource_2",
+				Type: "global",
 				Labels: map[string]string{
-					// Labels passed through with no changes
-					"source.label1": "value1",
+					"project_id": "123",
```
Comment (rghetia) [inline exporter/stackdriverexporter/resource_mapper_test.go:96]: For a moment I was wondering why project_id got converted even though the other mandatory label is missing. But the project_id is converted by default implementation.  Perhaps add this info in the comment.

**Review by rghetia — APPROVED** [review]
LGTM except for one minor change in the comment.

File: exporter/stackdriverexporter/resource_mapper_test.go (hunk 1 above)
Comment (nilebox) [reply exporter/stackdriverexporter/resource_mapper_test.go:97]: Added comments to clarify.

File: exporter/stackdriverexporter/config.go [hunk 2]
```diff
@@ -27,5 +27,20 @@ type Config struct {
...
+	SourceType string `mapstructure:"source_type"`
+	TargetType string `mapstructure:"target_type"`
+
+	LabelMappings []LabelMapping `mapstructure:"label_mappings"`
+}
+
+type LabelMapping struct {
+	SourceKey string `mapstructure:"source_key"`
+	TargetKey string `mapstructure:"target_key"`
+	Optional  bool   `mapstructure:"optional"`
```
Comment (x13n) [inline exporter/stackdriverexporter/config.go:45]: Can you document the semantics of this parameter? When would one set it to true?

Comment (nilebox) [reply exporter/stackdriverexporter/config.go:45]: It just signals that we can proceed with transformation when a particular label with a given `source_key` is missing.
I will add a comment.

Comment (nilebox) [reply exporter/stackdriverexporter/config.go]: Done.

**Pull request merged**